### PR TITLE
Cap max number of connections used by default

### PR DESCRIPTION
**Pull request by cevian**
Cap to 50 connections

### CODE REVIEW

**Comment by cevian** [comment]
Addresses #202

File: pkg/pgclient/client.go [hunk 1]
```diff
@@ -161,7 +166,7 @@ func (cfg *Config) GetNumConnections() (min int, max int, numCopiers int, err er
 
 	numCopiers = perProc * maxProcs
 	// we leave one connection per-core for non-copier usages
-	if numCopiers+maxProcs > max {
+	if numCopiers > max-maxProcs {
 		log.Warn("msg", fmt.Sprintf("had to reduce the number of copiers due to connection limits: wanted %v, reduced to %v", numCopiers, max/2))
 		numCopiers = max / 2
```
Comment (davidkohn88) [inline pkg/pgclient/client.go:171]: shouldn't this just be max-maxProcs? rather than max/2?

File: pkg/pgclient/client.go [hunk 2]
```diff
@@ -21,6 +21,8 @@ import (
 	"github.com/timescale/timescale-prometheus/pkg/util"
 )
 
+const maxDefaultConnections = 50
```
Comment (JLockerman) [inline pkg/pgclient/client.go:24]: I don't see any reason to have this up here instead of in the function

File: pkg/pgclient/client.go [hunk 3]
```diff
@@ -161,7 +166,7 @@ func (cfg *Config) GetNumConnections() (min int, max int, numCopiers int, err er
 
 	numCopiers = perProc * maxProcs
 	// we leave one connection per-core for non-copier usages
-	if numCopiers+maxProcs > max {
+	if numCopiers > max-maxProcs {
```
Comment (JLockerman) [inline pkg/pgclient/client.go:169]: if we're doing this, do you want to set the default `db-writer-connection-concurrency` back to `5`?

File: pkg/pgclient/client.go (hunk 1 above)
Comment (cevian) [reply pkg/pgclient/client.go:171]: no because maxProcs may be > max. Changed comment.

**Pull request merged**